### DISCUSSION TITLE
Rshell REPL works but minicom and pyterm do not

### DISCUSSION THREAD

**Discussion by MK27** [micropython/micropython#16138] [ESP32] [answered]
I know this maybe a bit of a nitpick, but I just got two new ESP32 WROOM boards (https://www.amazon.ca/dp/B0D4X3S1LQ?ref=ppx_yo2ov_dt_b_fed_asin_title); via `esptool`  they report as:

```
esptool.py v4.8.1
Serial port /dev/ttyUSB1
Connecting......
Chip is ESP32-D0WD-V3 (revision v3.1)
Features: WiFi, BT, Dual Core, 240MHz, VRef calibration in efuse, Coding Scheme None
Crystal is 40MHz
```

These are my first ESP32s; I've used the ESP8266 before.
I flashed v1.24.0 of the micropython firmware release from here:  https://micropython.org/download/ESP32_GENERIC/

     esptool  --port /dev/ttyUSB1 --chip esp32 --baud 115200 write_flash -fm dio -z 0x1000 ESP32_GENERIC-20241025-v1.24.0.bin

I ended up doing this multiple times because of the issue described below, and if I replaced `-fm dio`  and `--baud 115200` with `-fm qio` and `--baud 460800`, connecting to the REPL failed no matter what the tool, despite the fact that the install went without error.

After I flashed one board the first time, `minicom` would not show a REPL prompt and was totally unresponsive; it's what I prefer to use with micropython (mostly on Pi Pico and ESP8266 boards).  I have noticed in the past that sometimes when it when it would not work `pyterm.py` did.  In this case, I got a prompt but hitting enter did not work, the only thing it responded to was ctrl-C which simply yielded another prompt.

I've actually never used `rshell` for the REPL so did not think of this right away.  I tried the other board, same issue.  I tried using the next to last micropython (v1.23.0) instead, same issue.  I always used `erase_flash` before `write_flash`. I used 115200 as the explicit baud rate with both `minicom` and `pyterm`.

However, using `rshell` to check the filesystem worked fine; I could `cat /pyboard/boot.py` etc. (just 4 lines all commented out).  I  found out about the `repl` command and viola, there seems to be no problem.

So I'm happy, but I'm interested to know what explanation there is for this.  Why does `rshell` work when `minicom` does not?

**Comment by robert-hh** [comment, marked as answer]
>  Why does rshell work when minicom does not?

You have to set hardware flow control off, which is a little bit tricky with minicom.

**Reply by MK27** [reply]
Cheers -- did that via the config menu (ctrl-a, z, o, "Serial port setup", "Hardware flow control" -> off) and instantly got a working prompt.

**Reply by robert-hh** [reply]
That's what I meant with tricky. No simple command line option. I prefer picocom, which is smaller and simpler, or mpremote.
